### PR TITLE
Fix #51245 vertex tool topo crs mismatch

### LINKED ISSUE #51245: Topological editing does not work when project CRS is different than layer CRS
### What is the bug or the crash?

If you are trying to topologically edit a layer while your project crs is different than the layer crs, the vertex editor will only move one layer vertex and not both. 

### Steps to reproduce the issue

Create layer in a CRS, set project to different CRS. Split layer, then try moving the shared vertex. Only one will move.

### Versions

3.28.1 

### Supported QGIS version

- [x] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

### Additional context

Seems to have been fixed in 3.26 with [45357](https://github.com/qgis/QGIS/pull/45357) , but has reappeared in 3.28.1

### PR DESCRIPTION
**Pull request by Djedouas**
## Description

Fixes #51245 where topological editing with vertex tool does not work when layer CRS and project CRS are different.

### CODE REVIEW

**Comment by agiudiceandrea** [comment]
@Djedouas, there is a failing test.

**Comment by nyalldawson** [comment]
@Djedouas is this still a draft? If so, can we please close long-standing draft PRs to keep the queue clean?

**Comment by Djedouas** [comment]
Hi @nyalldawson 

I hoped for some help about the test integration… 
The code itself does work properly and fixes the issue. 

I plan to get back on my PR(s) soon, after having to work on other topics recently.

**Comment by Djedouas** [comment]
Ready for review!
Unrelated test error.

**Review by lbartoletti — APPROVED** [review]
tested together, LGTM and a nice simplification. Thanks!

**Comment by kannes** [comment]
Oh nice, thank you so much!

**Comment by agiudiceandrea** [comment]
Since this PR fixes a bug, shouldn't it be backported to both 3.40 and 3.34, or at least to 3.40?

File: src/app/vertextool/qgsvertextool.cpp [hunk 1]
```diff
@@ -1935,29 +1913,29 @@ void QgsVertexTool::buildDragBandsForVertices( const QSet<Vertex> &movingVertice
...
 
-QList<QgsPointLocator::Match> QgsVertexTool::layerVerticesSnappedToPoint( QgsVectorLayer *layer, const QgsPointXY &mapPoint )
+QgsPointLocator::MatchList QgsVertexTool::layerVerticesSnappedToPoint( QgsVectorLayer *layer, const QgsPointXY &mapPoint )
 {
   MatchCollectingFilter myfilter( this );
   QgsPointLocator *loc = canvas()->snappingUtils()->locatorForLayer( layer );
-  loc->nearestVertex( mapPoint, 0, &myfilter, true );
-  return myfilter.matches;
+  double tol = QgsTolerance::vertexSearchRadius( canvas()->mapSettings() );
+  return loc->verticesInRect( mapPoint, tol, &myfilter, true );
```
Comment (Djedouas) [inline src/app/vertextool/qgsvertextool.cpp:1921]: Should be changed to


```suggestion
  return loc->verticesInRect( mapPoint, 0, &myfilter, true );
```

or

```suggestion
  return loc->verticesInRect( mapPoint, 1e-8, &myfilter, true );
```

**Pull request merged**